.github/workflows/prototype-tests-gpu.yml

```diff
@@ -0,0 +1,79 @@
+# prototype-tests.yml adapted for self-hosted with gpu
+name: tests-gpu
+
+on:
+  pull_request:
+
+jobs:
+  prototype:
+    strategy:
+      fail-fast: false
+
+    runs-on: [self-hosted, linux.4xlarge.nvidia.gpu]
```

Comment:
As you are using the shared `linux.4xlarge.nvidia.gpu`, I think you could try the same approach as  PyTorch CI to make this works.  

* Install NVIDIA driver (nvidia-smi) and docker nvidia support https://github.com/pytorch/pytorch/blob/master/.github/workflows/_linux-test.yml#L63-L85
* Run the test in docker https://github.com/pytorch/pytorch/blob/master/.github/workflows/_linux-test.yml#L156, but you would need to provide the GPU_FLAG `--gpus all` for docker to correct pick up NVIDIA driver (nvidia-smi)

I have a feeling that `nvidia-smi` doesn't show up because the GPU_FLAG is missing https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container.  So you could also try to see if the following works.  You would still need to install NVIDIA driver first though because the AMI for `linux.4xlarge.nvidia.gpu` doesn't have it (I remember we track this issue somewhere)

```
container:
  image: pytorch/conda-builder:cuda116
  options: --gpus all
```